addons/mass_mailing_sms/models/res_users.py

```diff
@@ -1,77 +1,74 @@
-                                WHEN %(today)s::date - act.date_deadline::date > 0 Then 'overdue'
-                                WHEN %(today)s::date - act.date_deadline::date < 0 Then 'planned'
-                            END AS states
-                        FROM mail_activity AS act
-                        JOIN mailing_mailing AS m ON act.res_id = m.id
-                        WHERE act.res_model = 'mailing.mailing' AND act.user_id = %(user_id)s  
-                        GROUP BY m.mailing_type, states, act.res_model, act.res_id;
-                        """
-                self.env.cr.execute(query, {
-                    'today': fields.Date.context_today(self),
-                    'user_id': self.env.uid,
-                })
-                activity_data = self.env.cr.dictfetchall()
-                
-                user_activities = {}
-                for act in activity_data:
-                    if not user_activities.get(act['mailing_type']):
-                        if act['mailing_type'] == 'sms':
-                            module = 'mass_mailing_sms'
-                            name = _('SMS Marketing')
-                        else:
-                            module = 'mass_mailing'
-                            name = _('Email Marketing')
-                        icon = module and modules.module.get_module_icon(module)
-                        res_ids = set()
-                        user_activities[act['mailing_type']] = {
-                            'id': self.env['ir.model']._get('mailing.mailing').id,
-                            'name': name,
-                            'model': 'mailing.mailing',
-                            'type': 'activity',
-                            'icon': icon,
-                            'domain': [('active', 'in', [True, False])],
-                            'total_count': 0, 'today_count': 0, 'overdue_count': 0, 'planned_count': 0,
-                            'res_ids': res_ids,
-                            "view_type": view_type,
-                        }
-                    user_activities[act['mailing_type']]['res_ids'].add(act['res_id'])
-                    user_activities[act['mailing_type']]['%s_count' % act['states']] += act['count']
-                    if act['states'] in ('today', 'overdue'):
-                        user_activities[act['mailing_type']]['total_count'] += act['count']
+        activity_groups = super()._get_activity_groups()
+
+        # 1. removing mailing.mailing activity group
+        to_remove = next((g for g in activity_groups if g.get('model') == 'mailing.mailing'), None)
+        if not to_remove:
+            return activity_groups
+
+        activity_groups.remove(to_remove)
+
+        # 2. Re-fetch only mailing.mailing activities
+        search_limit = int(self.env['ir.config_parameter'].sudo().get_param('mail.activity.systray.limit', 1000))
+        mailing_activities = self.env["mail.activity"].search(
```

Comment:
Actually, now that we started to refactor it... it is quite strange to discard previously fetch activities, and fetch everything again.

What about adding 'activities' key for each model in returned structure ? That way you can use it here and split the group into two, using already fetch activities.

Global purpose is that overrides should not do custom queries, only split / order / customize groups based on given activities. That way we fetch once, and order in several addons.

You began to do it, but I think we can go further in the code cleaning :) .